sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+   options.AccessConditions.IfNoneMatch = Azure::ETag::Any();
+   blobClient.Upload(stream, options);
+   ```
+1. Use lease for more complex scenarios that invole multiple operations on the same resource.
+
+## What is the difference between `BlockBlobClient::Upload()` and `BlockBlobClient::UploadFrom()`? When should I use one vs the other?
+
+`BlockBlobClient::Upload()` takes a stream as parameter and uploads the stream as a block blob with exact one HTTP request.
+The blob created with this method doesn't have any blocks, which means functions like `StageBlock()`, `CommitBlockList()` or `GetBlockList()` don't apply here.
+You cannot append blocks to such blobs (You'll have to overwrite it with a new one).
+You want to use this one if you need precise control over SDK behavior at HTTP level.
+
+`BlockBlobClient::UploadFrom` takes a memory buffer or file name as parameter, splits the data in the buffer or file into smaller chunks intelligently or according to some parameters if provided,
+then upload the chunks with multiple threads.
+This one suits in most cases. You can expect higher throughput because the chunks are transferred concurrently. It's especially recommended if you need to transfer large blobs efficiently.
+
+## How to efficiently upload large amount of small blobs?
+
+Unfortunately this SDK doesn't provide a convenient way to upload many blobs or directory contents (files and sub-directories) with just one function call.
+You have to create multiple threads, traverse the directories by yourself and upload blobs one by one in each thread to speed up the transfer.
+Or you can use tools like [AzCopy](https://learn.microsoft.com/azure/storage/common/storage-ref-azcopy) or Data Movement Library.
+
+## How to ensure data integrity with transactional checksum?
+
+Generally speaking, TLS protocol includes checksum that's strong enough to detect accidental corruption or deliberate tampering.
+Another layer of checksum is usually not necessary if you're using HTTPS.
+
+If you really want it for whatever reason, for example, to detect corruptions before the data is written to the socket,
+you can leverage transactional checksum feature in the SDK.
+With this feature, you provide a pre-calculated MD5 or CRC64 checksum when calling an upload API,
+storage service will calculate checksum after it receives the data and compare with the one you provide,
+and fail the request if they don't match.
+Make sure you calculate the checksum early enough to cover the time when the corruption may happen.
+
+This functionality also works for download operations.
+Below is a code sample to use this feature.
+
+```C++
+// upload data with pre-calculated checksum
+Blobs::UploadBlockBlobOptions options;
+options.TransactionalContentHash = ContentHash();
+options.TransactionalContentHash.Value().Algorithm = HashAlgorithm::Crc64;
+options.TransactionalContentHash.Value().Value = crc64;  // CRC64 checksum of the data in Base64 encoding
```

Comment:
I think it will be cleaner to create a local `ContentHash` variable, set it's value and algorithm, and then set it to the `options.TransactionalContentHash` field which is `Nullable<T>`.

This avoids the `Value().Value` problem.

As an aside, we have a guideline going forward to avoid `Value` as a generic model field name for this reason.